src/result.cpp

```diff
@@ -116,30 +113,26 @@ read_embedded_file_entry(CacheEntryReader& reader,
                          const ResultFileMap* result_file_map,
                          FILE* dump_stream)
 {
-  uint8_t suffix_len;
-  reader.read(suffix_len);
-
-  char suffix[256];
-  reader.read(suffix, suffix_len);
+  FileType type;
+  reader.read(*reinterpret_cast<UnderlyingFileTypeInt*>(&type));
 
   uint64_t file_len;
   reader.read(file_len);
 
   bool content_read = false;
-  std::string suffix_str(suffix, suffix_len);
   if (dump_stream) {
     fmt::print(dump_stream,
-               "Embedded file #{}: {} ({} bytes)\n",
+               "Embedded file #{}: type {} ({} bytes)\n",
                entry_number,
-               suffix_str,
+               +static_cast<UnderlyingFileTypeInt>(type),
```

Comment:
It would be nice to print a human-readable string instead of (or in addition to) an integer here and in similar places.